pkgs/top-level/python-packages.nix

```diff
@@ -5389,6 +5389,10 @@ self: super: with self; {
 
   h5py = callPackage ../development/python-modules/h5py { };
 
+  h5py-numpy_2 = self.h5py.override {
+    numpy = numpy_2;
+  };
```

Comment:
On a 2nd thought, perhaps it'd be better to not propagate numpy in `h5py`, since:

```
$ nix why-depends --all --precise nixpkgs#python312Packages.{h5py,numpy}
/nix/store/mz66y00715zzxgzfsx415wr4nc8nxcbr-python3.12-h5py-3.11.0
└───nix-support/propagated-build-inputs: …/nix/store/jc6k07xdz70lfwbfiffz9m2wn6ag3699-python3.12-numpy-1.26.4 /nix/st…
    → /nix/store/jc6k07xdz70lfwbfiffz9m2wn6ag3699-python3.12-numpy-1.26.4
```